### PR TITLE
`azurerm_kubernetes_cluster`: Support migration of `service_principal` to `identity`

### LINKED ISSUE #11288: Support for switching from Service Principal to Managed Identity for azurerm_kubernetes_cluster without cluster recreation
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently, an AKS cluster with Service Principal can't migrate to Managed Identity without recreating the cluster from scratch.
But for updating Azure AD from Service Principal to Managed one, this works as "update in-place"

```hcl
~ role_based_access_control {
      # (1 unchanged attribute hidden)

    ~ azure_active_directory {
        ~ admin_group_object_ids = [
            + "********-****-****-****-b84c7e895b53",
          ]
        - client_app_id          = "********-****-****-****-34e3ae56d094" -> null
        ~ managed                = false -> true
        - server_app_id          = "********-****-****-****-1bc5e70c4750" -> null
        - server_app_secret      = (sensitive value)
          # (1 unchanged attribute hidden)
      }
  }
```

As I see in documentation: https://docs.microsoft.com/en-us/azure/aks/use-managed-identity#update-an-aks-cluster-to-managed-identities-preview
AKS has some process to migrate from SP to MI, and it in the Preview at the moment.
After GA of this feature, will this behavior be switched to update in-place?

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #11288

## TO DO
- [x] Cover edge cases
- [x] Docs
- [x] Run tests (only `servicePrincipal` tests)

## Acceptance Tests:
- [x] `TestAccKubernetesCluster_servicePrincipal` -> `FAIL  because I don't have a secondary principal specified`
- [x] `TestAccKubernetesCluster_servicePrincipalToSystemAssignedIdentity` -> `PASS`
- [x] `TestAccKubernetesCluster_servicePrincipalToUserAssignedIdentity` -> `PASS`

### CODE REVIEW

**Comment by aristosvo** [comment]
@mbfrahry Another AKS upgrade 😄

I hope you don't mind, would you be able to run the whole AccTests suite? I don't have the resources at the moment. Other than that, I'm not sure whether we should cover more edge cases, like:
- Users trying to change back from `identity` to `service_principal`
- Users updating `identity` blocks (not completely sure this won't be supported soon)

Do you think we should defend ourselves against these kind of edge cases?

File: website/docs/r/kubernetes_cluster.html.markdown [hunk 1]
```diff
@@ -92,9 +92,10 @@ In addition, one of either `identity` or `service_principal` blocks must be spec
 
 * `disk_encryption_set_id` - (Optional) The ID of the Disk Encryption Set which should be used for the Nodes and Volumes. More information [can be found in the documentation](https://docs.microsoft.com/en-us/azure/aks/azure-disk-customer-managed-keys).
 
-* `identity` - (Optional) An `identity` block as defined below. Changing this forces a new resource to be created.
+* `identity` - (Optional) An `identity` block as defined below.
```
Comment (mbfrahry) [inline website/docs/r/kubernetes_cluster.html.markdown:95]: I think we can put `One of either `identity` or `service_principal` must be specified.` in this message and then have the migration information in it's own separate note. It's a little different than how we normally do it but I'd like to make sure that we try and make it as clear as possible what a user needs to do when updating

File: website/docs/r/kubernetes_cluster.html.markdown [hunk 2]
```diff
@@ -160,7 +161,8 @@ resource "azurerm_kubernetes_cluster" "example" {
 
 * `service_principal` - (Optional) A `service_principal` block as documented below.
 
--> **NOTE:** One of either `identity` or `service_principal` must be specified.
+!> **NOTE:** One of either `identity` or `service_principal` must be specified. A migration scenario from `service_principal` to `identity` is supported. When upgrading `service_principal` to `identity`,
```
Comment (mbfrahry) [inline website/docs/r/kubernetes_cluster.html.markdown:164]: Same note here

**Review by mbfrahry — CHANGES_REQUESTED** [review]
This is looking good! I just have some suggestions about presenting the info in the docs

File: website/docs/r/kubernetes_cluster.html.markdown (hunk 1 above)
Comment (aristosvo) [reply website/docs/r/kubernetes_cluster.html.markdown]: Done, thanks!

File: website/docs/r/kubernetes_cluster.html.markdown (hunk 2 above)
Comment (aristosvo) [reply website/docs/r/kubernetes_cluster.html.markdown]: 👍🏽

**Review by mbfrahry — APPROVED** [review]
LGTM!
